### PR TITLE
Exposing enums on Realm constructor

### LINKED ISSUE #3365: Enums declared in types are missing implementations
## Goals
Use values from enums in code.

## Expected Results
```typescript
const Realm = require("realm");
assert(typeof Realm.ConnectionState === "object");
assert(typeof Realm.ConnectionState.Disconnected === "string");
```

## Actual Results
The enums (except `UpdateMode`) declared by the types are all undefined:

* `Realm.SessionStopPolicy`
* `Realm.OpenRealmBehaviorType`
* `Realm.OpenRealmTimeOutBehavior`
* `Realm.ConnectionState`
* `Realm.NumericLogLevel`

## Steps to Reproduce & Code Sample
1. Open up the Node.js REPL
1. Require Realm `const Realm = require("realm")`
1. Try accessing the enum `Realm.ConnectionState.Disconnected`
1. `Uncaught TypeError: Cannot read property 'Disconnected' of undefined` 

## Version of Realm and Tooling
* Realm JS SDK Version: ^10.0.1
* Node or React Native: Tested on Node.js (it's most likely also an issue in React Native).
* Client OS & Version: N/A
* Which debugger for React Native: None

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This closes #3365 and https://github.com/realm/realm-js/issues/5480#issuecomment-1442375770.

## ☑️ ToDos
* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 📝 Update `COMPATIBILITY.md`
* [x] 🚦 Tests (manually - locally)
* [x] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [x] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [x] 📱 Check the React Native/other sample apps work if necessary
* [x] 📝 Public documentation PR created or is not necessary
* [X] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: lib/extensions.js [hunk 1]
```diff
@@ -509,34 +509,51 @@ module.exports = function (realmConstructor) {
...
+
     realmConstructor.App.Sync.SessionState = {
       Invalid: "invalid",
       Active: "active",
       Inactive: "inactive",
     };
 
+    realmConstructor.SessionState = realmConstructor.App.Sync.SessionState;
+
     realmConstructor.App.Sync.ProgressDirection = {
```
Comment (takameyer) [inline lib/extensions.js:522]: This appears not to be set in `index.d.ts` on `App.Sync`

File: lib/extensions.js [hunk 2]
```diff
@@ -509,34 +509,51 @@ module.exports = function (realmConstructor) {
...
       Download: "download",
       Upload: "upload",
     };
 
+    realmConstructor.ProgressDirection = realmConstructor.App.Sync.ProgressDirection;
+
     realmConstructor.App.Sync.ProgressMode = {
```
Comment (takameyer) [inline lib/extensions.js:529]: Also not typed on `App.Sync`

File: lib/extensions.js [hunk 3]
```diff
@@ -509,34 +509,51 @@ module.exports = function (realmConstructor) {
...
       ReportIndefinitely: "reportIndefinitely",
       ForCurrentlyOutstandingWork: "forCurrentlyOutstandingWork",
     };
 
+    realmConstructor.ProgressMode = realmConstructor.App.Sync.ProgressMode;
+
     realmConstructor.App.Sync.ClientResyncMode = {
```
Comment (takameyer) [inline lib/extensions.js:536]: I think this should be `ClientResetMode`.  `ClientResyncMode` doesn't exist in the `.d.ts`

File: lib/extensions.js [hunk 4]
```diff
@@ -509,34 +509,51 @@ module.exports = function (realmConstructor) {
...
+    realmConstructor.ClientResyncMode = realmConstructor.App.Sync.ClientResyncMode;
+
     realmConstructor.App.Sync.SubscriptionsState = {
       Pending: "pending",
       Complete: "complete",
       Error: "error",
       Superseded: "superseded",
     };
+
+    realmConstructor.SubscriptionsState = realmConstructor.App.Sync.SubscriptionsState;
```
Comment (takameyer) [inline lib/extensions.js:551]: `SubscriptionState` is not typed on the `Realm` namespace

**Review by takameyer — CHANGES_REQUESTED** [review]
I was looking at the `index.d.ts` side by side with these changes and noticed a few inconsistancies.

File: lib/extensions.js (hunk 3 above)
Comment (kraenhansen) [reply lib/extensions.js:536]: Thanks :+1:

**Review by gagik — APPROVED** [review]

**Review by takameyer — APPROVED** [review]

**Pull request merged**